### DISCUSSION TITLE
Comparison function for numeric values

### DISCUSSION THREAD

**Discussion by dshopin** [moj-analytical-services/splink#723] [General]
Is there a way to make comparisons based on simple math calculations like absolute or relative difference? I looked at `distance_function_at_thresholds()`, but it seems to require some named function with arguments and this function should exist in the backend. I can create a UDF in Athena, of course, but is this the only way for now?

**Comment by RobinL** [comment]
Yes, if you write your own comparison levels as a dictionary you can define the comparison however you like in the sql_condition:

https://moj-analytical-services.github.io/splink/topic_guides/customising_comparisons.html#method-3-providing-the-spec-as-a-dictionary

E.g `abs(numeric_col_l - numeric_col_r) < 5`
